scalding-core/src/main/scala/com/twitter/scalding/Stats.scala

```diff
@@ -63,9 +64,30 @@ private[scalding] final case class GenericFlowPCounterImpl(fp: FlowProcess[_], s
   override def increment(amount: Long): Unit = fp.increment(statKey.group, statKey.counter, amount)
 }
 
+private[scalding] object HadoopFlowPCounterImpl {
+  @transient lazy val logger: Logger = LoggerFactory.getLogger(this.getClass)
+}
+
 private[scalding] final case class HadoopFlowPCounterImpl(fp: HadoopFlowProcess, statKey: StatKey) extends CounterImpl {
-  private[this] val cntr = fp.getReporter().getCounter(statKey.group, statKey.counter)
-  override def increment(amount: Long): Unit = cntr.increment(amount)
+  import HadoopFlowPCounterImpl.logger
+
+  private[this] val cntr: Option[Counter] = {
+    val reporter = fp.getReporter
```

Comment:
If we can skip all the logging maybe we can make it more idomatic/shorter? (is the logging useful/going to be read? 

```
Option(fp.getReporter).flatMap { reporter =>
   Option(reporter.getCounter(statKey.group, statKey.counter)).flatMap(identity)
}
```

Reply:
or even:
```scala
(for {
 rep <- Option(fp.getReporter)
 cnt <- Option(reporter.getCounter(statKey.group, statKey.counter))
} yield cnt).fold {
  logger.warn(s"Cannot increment counter(${statKey.group}, ${statKey.counter}) because HadoopFlowProcess.getReporter returned null")
}(())
```

Reply:
I have more logging in my company's fork of Scalding in order to try and track down the issue. I didn't want to include that full logging in this PR, but that makes this logging not useful. I'll remove the logging.

Reply:
Although a warning the first time would be useful so people can figure out why the counter was not incremented.